### ISSUE TITLE
Write errors to standard error instead of standard output

### ISSUE THREAD

**Issue by kai687** [adrienverge/yamllint#223]
Would it be possible to have errors written to `stderr` instead of `stdout`? I was surprised to find that although my test YAML file had errors (which were reported by `yamllint`), `stderr` was empty.

As it is now, I can include an extra check to make sure the return code is non-zero, and then print both `stdout` and `stderr`, but it would be nice to print errors where they belong, which in my experience is `stderr`.

**Comment by adrienverge** [comment]
Since yamllint is a *linter*, its normal output (written to `stdout`) are *linting problems*. Like most programs, yamllint uses `stderr` for runtime errors or bugs (unexpected encoding, missing file, crash...)

Moreover, what you ask would be a breaking change that would affect many, many users. What do you think?

**Comment by kai687** [comment]
Thanks for the explanation. The way you put it, it makes sense and I'd hate breaking changes. I can use `yamllint` in my pipeline checking for the return value instead of _just_ for content in `stderr`. Thanks again for responding.

**Comment by adrienverge** [comment]
You're welcome.

Another option for you could be to redirect `stdout` to `stderr` using Bash?

```shell
yamllint test.yaml >&2
```
